File: src/main/java/pw/vodes/rimuru/command/commands/CommandHelp.java
package pw.vodes.rimuru.command.commands;

import java.util.stream.Collectors;

import org.javacord.api.entity.message.embed.EmbedBuilder;
import org.javacord.api.event.message.MessageCreateEvent;

import pw.vodes.rimuru.Main;
import pw.vodes.rimuru.command.Command;
import pw.vodes.rimuru.command.CommandType;
import pw.vodes.rimuru.command.util.MultiPageEmbed;

public class CommandHelp extends Command {

	public CommandHelp() {
		super("Help", new String[] {"help", "h"}, CommandType.everyone);
	}

	@Override
	public void run(MessageCreateEvent event) {
		var embed = new EmbedBuilder().setTitle("Rimuru-Bot")
				.setAuthor(event.getMessageAuthor())
				.setThumbnail(Main.api.getYourself().getAvatar(4096))
				.addField("Version", Main.getVersion(), true)
				.addField("Written by", "<@129871096299126784>", true)
				.addField("Source code", "https://github.com/Vodes/Rimuru-Bot", false);
		
		var commands = new EmbedBuilder().setTitle("Commands")
				.setAuthor(event.getMessageAuthor());
		
		int current = 0;
		for(var cmd : Main.getCommandManager().getCommands().stream()
				.filter(c -> !c.getName().equalsIgnoreCase("help"))
				.sorted((c1, c2) -> c2.getName().compareToIgnoreCase(c1.getName()))
				.collect(Collectors.toList())) {
			commands.addField(cmd.getName() + (cmd.isEnabled() ? "" : " (disabled)"), cmd.getType().toString(), current == 2 ? false : true);
			current++;
			if(current > 2) 
				current = 0;
		}
		
		var multiEmbed = new MultiPageEmbed(embed, event.getMessageAuthor().asUser().get(), false, true);
		multiEmbed.addPage(commands);
		multiEmbed.sendMessage(event.getChannel());
	}

}

File: src/main/kotlin/pw/vodes/rimurukt/command/commands/CommandHelp.kt
package pw.vodes.rimurukt.command.commands

import org.javacord.api.entity.user.User
import org.javacord.api.event.message.MessageCreateEvent
import org.javacord.api.interaction.SlashCommandBuilder
import org.javacord.api.interaction.SlashCommandInteraction
import pw.vodes.rimurukt.Main
import pw.vodes.rimurukt.command.Command
import pw.vodes.rimurukt.command.Commands
import pw.vodes.rimurukt.components.MultiPageEmbed

class CommandHelp : Command("Help", arrayOf("help", "h"), slashCommandName = "help") {

    private fun getEmbed(user: User): MultiPageEmbed {
        val commit = Main.config.updateConfig.currentCommit
        val version = if (commit.isBlank()) Main.VERSION else
            "${Main.VERSION} [@${commit.subSequence(0, 6)}](https://github.com/Vodes/Rimuru-Bot/commit/$commit)"

        val commands = Commands.commands
            .filter { !it.name.equals("help", true) }
            .sortedBy { it.name }

        return MultiPageEmbed(user) {
            it.setTitle("Rimuru-Bot")
                .setAuthor(user)
                .setThumbnail(Main.api.yourself.getAvatar(4096))
                .addField("Version", version, true)
                .addField("Written by", "<@129871096299126784>", true)
                .addField("Source code", "https://github.com/Vodes/Rimuru-Bot/tree/rewrite-kotlin", false)
        }.addPage {
            it.setTitle("Commands").setAuthor(user)
            commands.forEachIndexed { index, command ->
                it.addField("${command.name}${if (command.enabled) "" else " (disabled)"}", command.type.toString(), index % 2 == 0)
            }
            it
        }
    }

    override fun run(event: MessageCreateEvent) {
        getEmbed(event.messageAuthor.asUser().get()).sendMessage(event)
    }

    override fun getSlashCommandBuilder(): SlashCommandBuilder? {
        return SlashCommandBuilder().setEnabledInDms(false).setName("help").setDescription("Displays available commands and general information on the bot.")
    }

    override fun runSlashCommand(interaction: SlashCommandInteraction) {
        getEmbed(interaction.user).sendAsResponse(interaction)
    }
}